src/Components/UpcomingTable/UpcomingRow.tsx

```diff
@@ -0,0 +1,130 @@
+    } else if (detail1 && detail2 && detail3) {
+      detail1Colspan = 2;
+      detail2Colspan = 2;
+      detail3Colspan = !childIsFullWidth ? 1 : 2;
+    }
+  }
+  return (
+    <Tbody key={`${repo.name}-${repo.type}-${repo.release}-${repo.date}`} 
+    isExpanded={isRepoExpanded}>
+      <Tr>
+        <Td
+          expand={{
+            rowIndex: rowIndex,
+            isExpanded: isRepoExpanded,
+            onToggle: () => setIsRepoExpanded(!isRepoExpanded),
+            expandId: "composable-expandable-example"
+          }}
+        />
+        <Td dataLabel={columnNames.name} modifier="truncate">
+            {repo.name}
+        </Td>
+        <Td dataLabel={columnNames.type} modifier="truncate">
+            {repo.type}
+        </Td>
+        <Td dataLabel={columnNames.release} modifier="truncate">
+            {repo.release}
+        </Td>
+        <Td dataLabel={columnNames.date} modifier="truncate">
+            {repo.date}
+        </Td>
+      </Tr>
+      {repo.details ? (
+        <Tr isExpanded={isRepoExpanded}>
+            {!childIsFullWidth ? <Td /> : null}
+            {repo.details.summary ? (
+            <Td
+                dataLabel="Summary"
+                noPadding={childHasNoPadding}
+                colSpan={detail1Colspan}
+            >
+                <ExpandableRowContent>
+                {repo.details.summary}
+                </ExpandableRowContent>
+            </Td>
+            ) : null}
+            {repo.details.potentiallyAffectedSystems ? (
+            <Td
+                dataLabel="Potentially affected systems"
+                noPadding={childHasNoPadding}
+                colSpan={detail2Colspan}
+            >
+                <ExpandableRowContent>
+                {repo.details.potentiallyAffectedSystems}
+                </ExpandableRowContent>
+            </Td>
+            ) : null}
+            {repo.details.trainingTicket ? (
+            <Td
+                dataLabel="Training ticket"
+                noPadding={childHasNoPadding}
+                colSpan={detail3Colspan}
+            >
+                <ExpandableRowContent>
+                {repo.details.trainingTicket}
+                </ExpandableRowContent>
+            </Td>
+            ) : null}
+            {repo.details.dateAdded ? (
+            <Td
+                dataLabel="Date added"
+                noPadding={childHasNoPadding}
+                colSpan={detail3Colspan}
+            >
+                <ExpandableRowContent>
+                {repo.details.dateAdded}
+                </ExpandableRowContent>
+            </Td>
+            ) : null}
+            {repo.details.lastModified ? (
+            <Td
+                dataLabel="Last modified"
+                noPadding={childHasNoPadding}
+                colSpan={detail3Colspan}
+            >
+                <ExpandableRowContent>
+                {repo.details.lastModified}
+                </ExpandableRowContent>
+            </Td>
```

Comment:
This is another great opportunity to either loop through `repo.details` or create a component that handles the rendering. You could possible create a Cell component that handles both this set of cells as well as the ones in the parent row. Then testing and refactoring become a breeze.